### PR TITLE
Refactor loadFromConf in separate methods

### PR DESCRIPTION
**Pull request by 42wim**
Trying to improve readability/maintainability, tackling methods with high `gocognit`

### CODE REVIEW

**Comment by lunny** [comment]
Some setting depends others, this need to be careful.

**Comment by 42wim** [comment]
> Some setting depends others, this need to be careful.

Yes I kept the same order as much as possible and moved some together that were depending on each other.

**Review by wxiaoguang — DISMISSED** [review]

**Comment by wxiaoguang** [comment]
@go-gitea/maintainers merge or not? I do not thing keeping it stale is safe.

**Comment by 42wim** [comment]
> Could we have a generic method named `newSection(sectionName string, sectionStruct any)`, so I think many method could be removed and some could start with the function and then have some special logic.

Ok, so I'll close this PR in favor of your suggestion someone will make?

File: modules/setting/setting.go [hunk 1]
```diff
@@ -1027,27 +1098,42 @@ func loadFromConf(allowEmpty bool, extraConfig string) {
...
+	sec := Cfg.Section("markdown")
+	if err := sec.MapTo(&Markdown); err != nil {
+		log.Fatal("Failed to map Markdown settings: %v", err)
+	}
+}
 
-	newRepository()
+func newAPI() {
+	sec := Cfg.Section("api")
+	if err := sec.MapTo(&API); err != nil {
```
Comment (delvh) [inline modules/setting/setting.go]:
```suggestion
	if err := Cfg.Section("api").MapTo(&API); err != nil {
```

File: modules/setting/setting.go [hunk 2]
```diff
@@ -1027,27 +1098,42 @@ func loadFromConf(allowEmpty bool, extraConfig string) {
 		}
 		log.Critical("You are running Gitea using the root user, and have purposely chosen to skip built-in protections around this. You have been warned against this.")
 	}
+}
 
-	SSH.BuiltinServerUser = Cfg.Section("server").Key("BUILTIN_SSH_SERVER_USER").MustString(RunUser)
-	SSH.User = Cfg.Section("server").Key("SSH_USER").MustString(SSH.BuiltinServerUser)
+func newMarkDown() {
+	sec := Cfg.Section("markdown")
+	if err := sec.MapTo(&Markdown); err != nil {
```
Comment (delvh) [inline modules/setting/setting.go]:
```suggestion
	if err := Cfg.Section("markdown").MapTo(&Markdown); err != nil {
```

File: modules/setting/setting.go [hunk 3]
```diff
@@ -1027,27 +1098,42 @@ func loadFromConf(allowEmpty bool, extraConfig string) {
...
-	} else if err = Cfg.Section("markdown").MapTo(&Markdown); err != nil {
-		log.Fatal("Failed to map Markdown settings: %v", err)
-	} else if err = Cfg.Section("admin").MapTo(&Admin); err != nil {
-		log.Fatal("Fail to map Admin settings: %v", err)
-	} else if err = Cfg.Section("api").MapTo(&API); err != nil {
-		log.Fatal("Failed to map API settings: %v", err)
-	} else if err = Cfg.Section("metrics").MapTo(&Metrics); err != nil {
+func newMetrics() {
+	sec := Cfg.Section("metrics")
+	if err := sec.MapTo(&Metrics); err != nil {
```
Comment (delvh) [inline modules/setting/setting.go]:
```suggestion
	if err := Cfg.Section("metrics").MapTo(&Metrics); err != nil {
```

File: modules/setting/setting.go [hunk 4]
```diff
@@ -1027,27 +1098,42 @@ func loadFromConf(allowEmpty bool, extraConfig string) {
...
 		log.Fatal("Failed to map Metrics settings: %v", err)
-	} else if err = Cfg.Section("camo").MapTo(&Camo); err != nil {
+	}
+}
+
+func newCamo() {
+	sec := Cfg.Section("camo")
+	if err := sec.MapTo(&Camo); err != nil {
```
Comment (delvh) [inline modules/setting/setting.go]:
```suggestion
	if err := Cfg.Section("camo").MapTo(&Camo); err != nil {
```

File: modules/setting/setting.go [hunk 5]
```diff
@@ -1027,27 +1098,42 @@ func loadFromConf(allowEmpty bool, extraConfig string) {
 		}
 		log.Critical("You are running Gitea using the root user, and have purposely chosen to skip built-in protections around this. You have been warned against this.")
 	}
+}
 
-	SSH.BuiltinServerUser = Cfg.Section("server").Key("BUILTIN_SSH_SERVER_USER").MustString(RunUser)
-	SSH.User = Cfg.Section("server").Key("SSH_USER").MustString(SSH.BuiltinServerUser)
+func newMarkDown() {
```
Comment (delvh) [inline modules/setting/setting.go:1102]: `newMarkdown`?

File: modules/setting/setting.go [hunk 6]
```diff
@@ -1004,12 +1028,59 @@ func loadFromConf(allowEmpty bool, extraConfig string) {
...
+	sec := Cfg.Section("ui")
+
+	if err := sec.MapTo(&UI); err != nil {
+		log.Fatal("Failed to map UI settings: %v", err)
+	}
+
+	UI.ShowUserEmail = sec.Key("SHOW_USER_EMAIL").MustBool(true)
+	UI.DefaultShowFullName = sec.Key("DEFAULT_SHOW_FULL_NAME").MustBool(false)
+	UI.SearchRepoDescription = sec.Key("SEARCH_REPO_DESCRIPTION").MustBool(true)
+	UI.UseServiceWorker = sec.Key("USE_SERVICE_WORKER").MustBool(false)
```
Comment (delvh) [inline modules/setting/setting.go:1050]: Isn't that covered by `MapTo` aready?

File: modules/setting/setting.go [hunk 7]
```diff
@@ -992,7 +1016,7 @@ func loadFromConf(allowEmpty bool, extraConfig string) {
 		}
 	}
 
-	zone := Cfg.Section("time").Key("DEFAULT_UI_LOCATION").String()
+	zone := sec.Key("DEFAULT_UI_LOCATION").String()
 	if zone != "" {
 		DefaultUILocation, err = time.LoadLocation(zone)
```
Comment (delvh) [inline modules/setting/setting.go:1020]:
```suggestion
		DefaultUILocation, err := time.LoadLocation(zone)
```

File: modules/setting/setting.go [hunk 8]
```diff
@@ -805,163 +977,15 @@ func loadFromConf(allowEmpty bool, extraConfig string) {
...
-	}
-	PasswordComplexity = make([]string, 0, len(cfgdata))
-	for _, name := range cfgdata {
-		name := strings.ToLower(strings.Trim(name, `"`))
-		if name != "" {
-			PasswordComplexity = append(PasswordComplexity, name)
-		}
-	}
+func newTime() {
+	var err error
```
Comment (delvh) [inline modules/setting/setting.go:983]:
```suggestion
```

File: modules/setting/setting.go [hunk 9]
```diff
@@ -592,52 +592,230 @@ func deprecatedSetting(oldSection, oldKey, newSection, newKey string) {
...
+	SSH.PerWriteTimeout = sec.Key("SSH_PER_WRITE_TIMEOUT").MustDuration(PerWriteTimeout)
+	SSH.PerWritePerKbTimeout = sec.Key("SSH_PER_WRITE_PER_KB_TIMEOUT").MustDuration(PerWritePerKbTimeout)
+
+	SSH.BuiltinServerUser = Cfg.Section("server").Key("BUILTIN_SSH_SERVER_USER").MustString(RunUser)
+	SSH.User = Cfg.Section("server").Key("SSH_USER").MustString(SSH.BuiltinServerUser)
+}
+
+func newOauth2() {
+	sec := Cfg.Section("oauth2")
+	if err := sec.MapTo(&OAuth2); err != nil {
```
Comment (delvh) [inline modules/setting/setting.go]:
```suggestion
	if err := Cfg.Section("oauth2").MapTo(&OAuth2); err != nil {
```

**Comment by wxiaoguang** [comment]
It's a pity that this PR became stale for so long time. Beside closing it I can not find a new ending for it .....

**Pull request closed without merging**